### PR TITLE
[Pytx] vPDQ index benchmark

### PR DESCRIPTION
**Pull request by tigerchensh**
Summary
---------

Benchmark on vPDQ with brute_force, vPDQ index with faiss and faiss search only. The difference between index and faiss is that vPDQ index stores and translates lots of data during building. Benchmark mark runs on MacBook Pro (16-inch, 2021), Apple M1 Pro, 32 GB Ram and the result is in README.MD.

### CODE REVIEW

File: python-threatexchange/benchmarks/benchmark_vpdq_index.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    ap.add_argument(
+        "--video-frames",
+        "-v",
+        type=int,
+        default=500,
+        help="How many frames in each video",
+    )
+    ap.add_argument(
+        "--dataset-size",
+        "-n",
```
Comment (Dcallies) [inline python-threatexchange/benchmarks/benchmark_vpdq_index.py:130]: n corresponding the number of videos, and `-v` also existing might be confusing. Would suggest either:
1. Using `-n` for video and `-m` for frames
2. Using `-v` for videos and `-f` for average frames

**Review by Dcallies — COMMENTED** [review]
This looks roughly like what I would expect, and would accept this if you are done with it - feel free to convert to real PR

File: python-threatexchange/benchmarks/benchmark_vpdq_index.py (hunk 1 above)
Comment (tigerchensh) [reply python-threatexchange/benchmarks/benchmark_vpdq_index.py]: Took this option "Using -v for videos and -f for average frames"

File: python-threatexchange/benchmarks/benchmark_vpdq_index.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    VPDQ_QUALITY_THRESHOLD,
+    VPDQ_DISTANCE_THRESHOLD,
+)
+
+import typing as t
+import random
+from threatexchange.extensions.vpdq.vpdq_index import VPDQIndex
+
+
+Oint = t.Optional[int]
```
Comment (BarrettOlson) [inline python-threatexchange/benchmarks/benchmark_vpdq_index.py:26]: nit: is this used anywhere?

File: python-threatexchange/benchmarks/benchmark_vpdq_index.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    ap.add_argument(
+        "--jitter-noise",
+        "-j",
+        type=int,
+        default=50,
+        help="How many frames varies between each video in uniformal distribution",
+    )
+    ap.add_argument(
+        "--dataset-size",
+        "-v",
```
Comment (BarrettOlson) [inline python-threatexchange/benchmarks/benchmark_vpdq_index.py:146]: nit: I will always read `-v` as verbose

**Review by BarrettOlson — APPROVED** [review]
Nice! had a few nits, nothing blocking

**Pull request merged**